### PR TITLE
Add unicode path support

### PR DESCRIPTION
**Pull request by EXtremeExploit**
# Testing ( Windows )
## Runtimes
*  [x] ~~CAN IT RUN CRYSIS?~~
* [x] Change pref path to one with unicode characters
  * [x] Does it work correctly with games/emu?
  * [x] Can you change it back? / reset it?
* [x] Run the emu in a path with unicode characters
# Testing ( Linux )
## Runtimes
*  [x] ~~CAN IT RUN CRYSIS?~~
* [x] Change pref path to one with unicode characters
  * [x] Does it work correctly with games/emu?
  * [x] Can you change it back? / reset it?
* [x] Run the emu in a path with unicode characters
# Testing ( MacOS )
## Runtimes
*  [x] ~~CAN IT RUN CRYSIS?~~
* [ ] Change pref path to one with unicode characters
  * [ ] Does it work correctly with games/emu?
  * [ ] Can you change it back? / reset it?
* [ ] Run the emu in a path with unicode characters

# Splitted work:
#957 
#953 
#962 

* Thx to @Zangetsu38 for helping me fixing the PR history and merging this

### CODE REVIEW

File: vita3k/gui/src/pkg_install_dialog.cpp [hunk 1]
```diff
@@ -115,24 +121,24 @@ void draw_pkg_install_dialog(GuiState &gui, HostState &host) {
             ImGui::Separator();
             ImGui::Spacing();
             ImGui::Checkbox("Delete the pkg file?", &delete_pkg_file);
-            if (work_path)
+            if (work_path != std::u16string())
```
Comment (sunho) [inline vita3k/gui/src/pkg_install_dialog.cpp:124]: Try if (work_path != u"")

File: vita3k/gui/src/pkg_install_dialog.cpp [hunk 2]
```diff
@@ -115,24 +121,24 @@ void draw_pkg_install_dialog(GuiState &gui, HostState &host) {
...
-                    fs::remove(fs::path(work_path));
+                    fs::remove(fs::path(string_utils::utf_to_wide(string_utils::utf16_to_utf8(pkg_path))));
                     delete_work_file = false;
                 }
                 update_notice_info(gui, host, "content");
                 if (host.app_category == "gd")
                     refresh_app_list(gui, host);
-                pkg_path = nullptr;
-                work_path = nullptr;
+                pkg_path = std::u16string();
```
Comment (sunho) [inline vita3k/gui/src/pkg_install_dialog.cpp:140]: Same

File: vita3k/io/src/io.cpp [hunk 3]
```diff
@@ -87,6 +88,7 @@ SpaceInfo get_space_info(const VitaIoDevice device, const std::string &vfs_path,
 // ****************************
 
 bool init(IOState &io, const fs::path &base_path, const fs::path &pref_path, bool redirect_stdio) {
+    //std::wstring base_path = string_utils::utf_to_wide(_base_path.string());
```
Comment (IllusionMan1212) [inline vita3k/io/src/io.cpp:91]: is this needed?

**Review by IllusionMan1212 — APPROVED** [review]
good work, we'll wait and see if @sunho can test on mac in the next few days, otherwise i'll just merge

File: vita3k/io/src/io.cpp (hunk 3 above)
Comment (EXtremeExploit) [reply vita3k/io/src/io.cpp:91]: No, its not, my bad

**Comment by sunho** [comment]
If it's okay in linux, I expect it to be okay in mac. I think it's fine to merge this. This pr has been here for a month already. If someone finds the regression in mac, please report!!

**Pull request merged**